File: public/cses/sorting-and-searching/concert-tickets/ub_approach/concert-tickets.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // setup
    // setup-start
#ifndef ONLINE_JUDGE
    freopen("../test_cases/1/input.txt", "r", stdin);
    freopen("../test_cases/1/output.txt", "w", stdout);
#endif
    // setup-end

    int n, m;
    cin >> n >> m;
    multiset<int> h;
    vector<int> t(m);
    for (int i = 0; i < n; i++) {
        int v;
        cin >> v;
        h.insert(v);
    }
    for (int i = 0; i < m; i++) {
        cin >> t[i];
    }
    for (int v : t) {
        auto it = h.upper_bound(v);

        if (it != h.begin()) {
            it--;
            cout << *it << endl;
            h.erase(it);
        } else {
            cout << -1 << endl;
        }
    }
}

File: public/cses/sorting-and-searching/concert-tickets/ub_approach/concert-tickets.py
# setup-start
import os
import sys
script_directory = os.path.dirname(os.path.realpath(__file__))
parent_directory = os.path.dirname(script_directory)
input_path = os.path.join(parent_directory, 'test_cases/1/input.txt')
output_path = os.path.join(parent_directory, 'test_cases/1/output.txt')
if os.path.exists(input_path):
    sys.stdin = open(input_path, 'r')
    try:
        sys.stdout = open(output_path, 'w')
    except Exception as e:
        pass
# setup-end


def main():
    import bisect

    # Read inputs
    n, m = map(int, input().split())
    h = sorted(map(int, input().split()))
    t = list(map(int, input().split()))

    for v in t:
        # Find the index of the first element greater than v
        idx = bisect.bisect_right(h, v)
        if idx == 0:
            print(-1)
        else:
            # Take the largest element <= v
            chosen = h[idx - 1]
            print(chosen)
            # Remove it from the list
            h.pop(idx - 1)


main()
